Source/Core/DolphinQt2/Config/Graphics/PostProcessingConfigWindow.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <QDialog>
+
+#include "VideoCommon/PostProcessing.h"
+
+class EnhancementsWidget;
+class QCheckBox;
+class QWidget;
+class QTabWidget;
+class QGridLayout;
+class QDialogButtonBox;
+class QLabel;
+class QSlider;
+
+class PostProcessingConfigWindow final : public QDialog
+{
+  Q_OBJECT
+public:
+  explicit PostProcessingConfigWindow(EnhancementsWidget* parent, const std::string& shader);
+  ~PostProcessingConfigWindow();
+
+private:
+  class ConfigGroup final
+  {
+  public:
+    ConfigGroup(const PostProcessingShaderConfiguration::ConfigurationOption* config_option);
+
+    const std::string& GetGUIName() const noexcept;
+    const std::string& GetParent() const noexcept;
+    const std::string& GetOptionName() const noexcept;
+    void AddSubGroup(std::unique_ptr<ConfigGroup>&& subgroup);
+    bool HasSubGroups() const noexcept;
+    const std::vector<std::unique_ptr<ConfigGroup>>& GetSubGroups() const noexcept;
+    unsigned int AddWidgets(PostProcessingConfigWindow* const parent, QGridLayout* const grid,
```

Comment:
These `const`s can be removed. value-wise const only applies in the definition of the function, it doesn't do anything in a declaration.